src/Build/Instance/RunningObjectTable.cs

```diff
@@ -14,34 +16,29 @@ namespace Microsoft.Build.Execution
     /// <remarks>
     /// See https://docs.microsoft.com/en-us/windows/desktop/api/objidl/nn-objidl-irunningobjecttable.
     /// </remarks>
-    internal class RunningObjectTable : IDisposable, IRunningObjectTableWrapper
+    internal class RunningObjectTable : IRunningObjectTableWrapper
```

Comment:
If `IRunningObjectTableWrapper` is an internal interface, and if the callers are or could be async, consider revising this interface to define async methods so you don't have to block the caller's thread while waiting for another thread.